src/test/java/com/google/devtools/build/lib/starlark/StarlarkRuleContextTest.java

```diff
@@ -2626,6 +2626,35 @@ public void testSpawnActionInterface() throws Exception {
     assertThat((Boolean) result).isTrue();
   }
 
+  @Test
+  public void testSpawnActionInterfaceEnvInherit() throws Exception {
+    useConfiguration("--action_env=MY_VAR");
```

Comment:
It feels wrong to write tests/asserts against the value of inherited_env in a Starlark test.  

Aren't all such tests going to be environment dependent and will fail if the variable isn't set? What am I missing? Will the tests always check for absence of some values?

Reply:
I would use this together with an analysis test transition on `--action_env` to ensure the variables I'm checking for are there. You can't check for all elements, but contains checks will be hermetic.

An alternative would be to assert on the value of `use_default_shell_env` passed to `run{_shell}` directly, but that feels like depending on an implementation detail to me.

Reply:
The first option sound like you're testing `--action_env` flag and not the rule.

Second option sound better, but also sound like `use_default_shell_env` is broken :)

Can you describe the requirement of the rule, that you need to test?

Reply:
Since the interaction between `env` and `use_default_shell_env` is fixed now and can be relied upon, I guess that what I would want to test is really that `use_default_shell_env` is enabled on the rule. I'll close this PR.